server/util.go

```diff
@@ -68,3 +72,24 @@ func secondsToDuration(seconds float64) time.Duration {
 	ttl := seconds * float64(time.Second)
 	return time.Duration(ttl)
 }
+
+// Parse a host/port string with an optional default port
+func parseHostPort(hostPort string, defaultPort string) (host string, port int, err error) {
```

Comment:
I would change `defaultPort` to an int here (as described in previous comments).